### PR TITLE
Fix hard-coded Python paths in CheckHealth.

### PR DESCRIPTION
**Pull request by vext01**
I've tested the Python provider path, but not the curl one (please check).

Before:
```
## Python 2 provider (optional)
  - INFO: `g:python_host_prog` is not set.  Searching for python2.7 in the environment.
  - INFO: Executable: /usr/local/bin/python2.7
  - ERROR: Command error -1: python -c from neovim import VERSION as v; print("{}.{}.{}{}".format(v.major, v.minor, v.patch, v.prerelease))
  - INFO: Python2 version: 2.7.13
  - INFO: python2.7-neovim version: 0.1.13
  - SUCCESS: Latest python2.7-neovim is installed: 0.1.13
```

After:
```
## Python 2 provider (optional)
  - INFO: `g:python_host_prog` is not set.  Searching for python2.7 in the environment.
  - INFO: Executable: /usr/local/bin/python2.7
  - INFO: Python2 version: 2.7.13
  - INFO: python2.7-neovim version: 0.1.13
  - SUCCESS: Latest python2.7-neovim is installed: 0.1.13
```

Thanks

### CODE REVIEW

File: runtime/autoload/health/provider.vim [hunk 1]
```diff
@@ -91,7 +91,7 @@ function! s:download(url) abort
   if executable('curl')
     let rv = s:system(['curl', '-sL', a:url], '', 1, 1)
     return s:shell_error ? 'curl error with '.a:url.': '.s:shell_error : rv
-  elseif executable('python')
+  elseif executable(a:python)
```
Comment (justinmk) [inline runtime/autoload/health/provider.vim]: Parameter needs to be added and passed

**Review by justinmk — CHANGES_REQUESTED** [review]

File: runtime/autoload/health/provider.vim (hunk 1 above)
Comment (vext01) [inline runtime/autoload/health/provider.vim]: Sorry, I'm not sure what you mean. (I don't really know vimscript)

Comment (vext01) [inline runtime/autoload/health/provider.vim]: oh, i see, one moment.

Comment (vext01) [inline runtime/autoload/health/provider.vim]: Actually, that's quite annoying to pass around, isn't it?

File: runtime/autoload/health/provider.vim [hunk 2]
```diff
@@ -101,12 +101,12 @@ function! s:download(url) abort
           \response = urlopen('".a:url."')\n
           \print(response.read().decode('utf8'))\n
           \"
-    let rv = s:system(['python', '-c', script])
+    let rv = s:system([a:python, '-c', script])
     return empty(rv) && s:shell_error
-          \ ? 'python urllib.request error: '.s:shell_error
+          \ ? a:python + 'urllib.request error: '.s:shell_error
```
Comment (ZyX-I) [inline runtime/autoload/health/provider.vim]: VimL is not using `+` for concatenation.

**Review by ZyX-I — CHANGES_REQUESTED** [review]

File: runtime/autoload/health/provider.vim (hunk 2 above)
Comment (brcolow) [inline runtime/autoload/health/provider.vim]: (It uses a dot, or period, for string concat)

**Comment by justinmk** [comment]
Going to fix/merge this now

File: runtime/autoload/health/provider.vim (hunk 1 above)
Comment (justinmk) [inline runtime/autoload/health/provider.vim]: Actually, I reverted this change. `curl` is the main expectation, `python` is a fallback. This is only a download utility, not related to general python env sniffing.

**Comment by justinmk** [comment]
merged

**Pull request closed without merging**